JuliaInterface/gap/adapter.gi

```diff
@@ -50,6 +37,66 @@ Perform([[ "IsJuliaObject", "IsJuliaObject" ],
     end);
 
 
+#
+# lists
+#
+InstallOtherMethod( \[\],
+    [ "IsJuliaObject", "IsPosInt and IsSmallIntRep" ],
+    function( obj, i )
+      return Julia.Base.getindex( obj, i );
+    end );
+
+InstallOtherMethod( \[\]\:\=,
+    [ "IsJuliaObject", "IsPosInt and IsSmallIntRep", "IsObject" ],
+    function( obj, i, val )
+      return Julia.Base.setindex\!( obj, val, i );
+    end );
+
+#
+# "matrices" / lists of lists
+#
+InstallOtherMethod( \[\],
+    [ "IsJuliaObject", "IsPosInt and IsSmallIntRep",
+                       "IsPosInt and IsSmallIntRep" ],
+    function( obj, i, j )
+      return Julia.Base.getindex( obj, i, j );
+    end );
+
+InstallOtherMethod( \[\]\:\=,
+    [ "IsJuliaObject", "IsPosInt and IsSmallIntRep",
+                       "IsPosInt and IsSmallIntRep", "IsObject" ],
+    function( obj, i, j, val )
+      return Julia.Base.setindex\!( obj, val, i, j );
+    end );
+
+#
+# access to fields and properties
+#
+BindGlobal("_JL_RNAM_TO_JULIA_SYMBOL_CACHE", rec());
```

Comment:
Guess this is an optimization to safe you looking the time looking up the string which corresponds to an `rnam`?

Reply:
Yes, this is an optimization.